src/main/java/com/ays/backend/user/service/impl/UserServiceImpl.java

```diff
@@ -65,4 +68,32 @@ public Optional<User> findByUsername(String username) {
     public boolean existsByUsername(String username) {
         return userRepository.existsByUsername(username);
     }
+
+    public Page<UserDTO> getAllUsers(Pageable pageable) {
+        return userRepository.findAll(pageable).map(
+                this::userToUserDTO
+        );
+    }
+
+    @Override
+    public UserDTO getUserById(Long id) {
+        return userRepository.findById(id).map(
+                this::userToUserDTO
+        ).orElseThrow(() -> new UserNotFoundException("User with id " + id + " not found"));
+    }
+
+    public UserDTO userToUserDTO(User user) {
+        return UserDTO.builder()
+                .username(user.getUsername())
+                .firstName(user.getFirstName())
+                .lastName(user.getLastName())
+                .userRole(UserRole.getById(user.getUserRole().ordinal()))
+                .countryCode(user.getCountryCode())
+                .lineNumber(user.getLineNumber())
+                .userStatus(user.getStatus())
+                .email(user.getEmail())
+                .lastLoginDate(user.getLastLoginDate())
+                .build();
+    }
```

Comment:
We can move this method into the `UserDTO` class and name it from for convenience.

Reply:
I just moved this method to `UserDTO`.

Reply:
It doesn't seem to be moved, did you forget to push it?

Reply:
@halilturkoglucs gave a comment on this pull request. He said to use a mapstruct instead of custom mapper in the long term. That's why I revised the method again.